packages/engine-ui/src/components/CreateBooking.js

```diff
@@ -38,18 +38,35 @@ const CreateBooking = ({ onSubmit }) => {
   const history = useHistory()
   const [isFinished, setIsFinished] = React.useState(false)
   const [formState, setState] = React.useState(initialState)
+  const [showErrorMessage, setErrorState] = React.useState({
+    pickup: false,
+    delivery: false,
+  })
   const setUIState = stores.ui((state) => state.dispatch)
 
   hooks.useFormStateWithMapClickControl('pickup', 'delivery', setState)
 
+  const validateAddresses = () => {
+    const pickup = !formState.pickup.lat || !formState.pickup.lon
+    const delivery = !formState.delivery.lat || !formState.delivery.lon
+
+    if (pickup || delivery) {
+      setErrorState((showErrorMessage) => ({
+        ...showErrorMessage,
+        pickup,
+        delivery,
+      }))
+
+      return false
+    }
+
+    return true
+  }
+
```

Comment:
Maybe move the `setErrorState(...)` outside of the validation function? Since you check further down `if (!validateAddresses())`,  I think it's more readable if you actually invoke the "consequences" of that condition there too.

Reply:
ifix